rbkit-lib/ui/cpuview.cpp

```diff
@@ -0,0 +1,62 @@
+#include "cpuview.h"
+#include "ui_cpuview.h"
+#include <QDebug>
+#include "model/cpustorage.h"
+
+CpuView::CpuView(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::CpuView)
+{
+    callGraphModel = new QStandardItemModel;
+    flatGraphModel = new QStandardItemModel;
+
+    ui->setupUi(this);
+
+    connect(this,
+            SIGNAL(fillCallGraph(QStandardItemModel*)),
+            RBKit::CpuStorage::getStorage().data(),
+            SLOT(fillCallGraphModel(QStandardItemModel*)));
+
+    connect(this,
+            SIGNAL(fillFlatProfile(QStandardItemModel*)),
+            RBKit::CpuStorage::getStorage().data(),
+            SLOT(fillFlatProfileModel(QStandardItemModel*)));
+
+    // generating both the models
+    emit fillCallGraph(callGraphModel);
+    emit fillFlatProfile(flatGraphModel);
+
+    // column headers for each model
+    QStringList headers;
+    headers.append("Methods");
+    headers.append("Self");
+    headers.append("Total");
+    this->callGraphModel->setHorizontalHeaderLabels(headers);
+    this->flatGraphModel->setHorizontalHeaderLabels(headers);
+
+    // setting callgraph as default view
+    updateCpuModel(callGraphModel);
+}
+
+CpuView::~CpuView()
+{
+    qDebug() << "++++++++++++++cpu view destroyed+++++++++++++++++";
+    delete ui;
```

Comment:
i was using it initially
but the problem i experienced was that, after `chartingTab.addTab(cpuviewprt.data(),"some title")`
the view got deleted, even when the above line was successful
that is why i used that `+++cpu view+++` debug message to look into it
it seems that if i create a single shared pointer in `centralwidget.h` file and add it to the tab, it persists, but creating new sharedpointer in [newCpuView](https://github.com/code-mancers/rbkit-client/blob/cpu-treeview/rbkit-lib/ui/centralwidget.cpp#L177) makes it persists only within the scope of `newCpuView` function, and gets destroyed just after that